### ISSUE TITLE
`with_genesis_config` not support `u128` type

### ISSUE THREAD

**Issue by AurevoirXavier** [paritytech/polkadot-sdk#4853]
`with_genesis_config` requires a `serde_json::Value` type. But it only supports up to `u64`.

But I think most chains are using `u128` as their balance type.

https://substrate.stackexchange.com/questions/11100/number-out-of-range-on-configuring-balances/11522#11522

**Comment by skunert** [comment]
I think this was discussed before in https://github.com/paritytech/polkadot-sdk/issues/2963 and should have been fixed in https://github.com/paritytech/polkadot-sdk/pull/2987. @michalkucharczyk Have not looked deeply, but probably we should enable this feature in `sc-chain-spec` too?

**Comment by AurevoirXavier** [comment]
> I think this was discussed before in #2963 and should have been fixed in #2987. @michalkucharczyk Have not looked deeply, but probably we should enable this feature in `sc-chain-spec` too?

Well, I don't think that's actually a fix on the user side. We still need to construct that JSON on our end and then pass it to the `with_genesis_config` function.

**Comment by michalkucharczyk** [comment]
Add this feature (`arbitrary_precision`):
https://github.com/serde-rs/json/blob/24d868f4e9be428afa1c744f8218a32660a1e0bf/Cargo.toml#L69-L75

here:
https://github.com/XcavateBlockchain/Xcavate_Node/blob/2868ef7a53bd6640d32e7030f047fa58bef7d924/node/Cargo.toml#L20

This should fix the problem.

**Comment by AurevoirXavier** [comment]
> This should fix the problem.

But should we should left this to the downstream?

What if make this function accept a `D: DeserializeOwned` or using function chain to add another patch like `.with_genesis_patch(BalancesConfig { balances: vec![(.., ..)] })`.

**Comment by michalkucharczyk** [comment]
From what I understand, the creation of JSON failed. Not deserialization on runtime side. (some `unwrap` that is inside `json!` macro).

**Comment by michalkucharczyk** [comment]
Basically this program fails:
```toml
[package]
name = "xxx"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
serde_json = {version = "1.0.117", default-features = false, features = ["alloc"] }
```

```rust
fn main() {
	serde_json::json!({
		"balances": {
			"balances": 1u128 << 112
		}
	});
}
```

**Comment by michalkucharczyk** [comment]
> I think this was discussed before in #2963 and should have been fixed in #2987. @michalkucharczyk Have not looked deeply, but probably we should enable this feature in `sc-chain-spec` too?

You mean that features shall be merged by cargo? Yeah, we could add this, as `sc-chain-spec` is on the node side. But I am not 100% sure how features eventually will be resolved.

BTW: if this genesis config was defined on the runtime side (as [preset](https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/reference_docs/chain_spec_genesis/index.html#genesis-state-presets-for-runtime)) then this would not be a problem.
